numerics/polynomial_evaluators.hpp

```diff
@@ -11,21 +11,38 @@ namespace internal_polynomial_evaluators {
 using quantities::Derivative;
 using quantities::Square;
 
-// The structs below are only needed because the language won't let us have (1)
-// a partial specialization of a function (2) an explicit specialization of a
-// member function template in an unspecialized class template.  Sigh.
+template<typename Value, typename Argument, int degree, bool allow_fma>
+struct EstrinEvaluator;
+template<typename Value, typename Argument, int degree, bool allow_fma>
+struct HornerEvaluator;
+
+namespace exported {
```

Comment:
Would it work to close `internal_polynomial_evaluators` here and reopen it near line 33?  That might avoid all the references to `exported`.